proxy.py

```diff
@@ -402,62 +496,192 @@ def __init__(self, activation_func):
     def forward(self, input):
         return self.activation(input)
 
+
 class MLP(nn.Module):
-    def __init__(self, config):
-        super().__init__()
-        # initialize constants and layers
+    def __init__(
+        self,
+        config,
+        transformerCall=False,
+        in_dim=512,
+        hidden_layers=[512, 512],
+        dropout_prob=0.0,
+    ):
+
+        super(MLP, self).__init__()
         self.config = config
-        act_func = "gelu"
+        self.device = config.device
+        act_func = "relu"
 
-        # Architecture
         self.input_max_length = self.config.env.max_len
-        self.input_classes = self.config.env.dict_size + 1  # +1 for eos
-        self.filters = 156
-        self.layers = 8
-
-        self.init_layer_depth = int(
-            (self.input_classes) * (self.input_max_length + 1)
-        )  # +1 for eos token
-
-        # build input layers
-        self.initial_layer = nn.Linear(
-            int(self.init_layer_depth), self.filters) 
-        self.initial_activation = Activation(act_func)
-        # output layer
-        self.output_layer = nn.Linear(self.filters, 1, bias=False)
-
-        # build hidden layers
-        self.lin_layers = []
-        self.activations= []
-        #self.norms = []  
-        self.dropouts= []
-
-        for i in range(self.layers):
-            self.lin_layers.append(
-                nn.Linear(self.filters, self.filters)
-            )
-            self.activations.append(
-                Activation(act_func)
-            )
-            #self.norms.append(nn.BatchNorm1d(self.filters))
-            self.dropouts.append(nn.Dropout(p = self.config.proxy.training.dropout))
+        self.input_classes = self.config.env.dict_size
+        self.out_dim = 1
+        self.transformerCall = transformerCall
 
-        # initialize module lists
-        self.lin_layers = nn.ModuleList(self.lin_layers)
-        self.activations = nn.ModuleList(self.activations)
-        #self.norms = nn.ModuleList(self.norms)
-        self.dropouts = nn.ModuleList(self.dropouts)
+        if self.transformerCall == False:
+            self.hidden_layers = [1024, 1024, 1024, 1024, 1024]
```

Comment:
It would be nice if the layer dimensionality could be selected via configuration, but not critical to add before merging.